src/integrations/jenkins_integration.rs

```diff
@@ -0,0 +1,180 @@
+use failure::Error;
+use integrations::jenkins_response::*;
+use network::{get_basic_credentials, get_url_response};
+use remote_status::RemoteStatus;
+use reqwest::header::{Authorization, Headers};
+use RemoteIntegration;
+
+pub struct JenkinsIntegration {
+    r: u16,
+    g: u16,
+    b: u16,
+    username: String,
+    password: String,
+    base_url: String,
+}
+
+impl JenkinsIntegration {
+    pub fn new(
+        r: u16,
+        g: u16,
+        b: u16,
+        username: &str,
+        password: &str,
+        base_url: &str,
+    ) -> JenkinsIntegration {
+        JenkinsIntegration {
+            r: r,
+            g: g,
+            b: b,
+            username: username.to_string(),
+            password: password.to_string(),
+            base_url: base_url.to_string(),
+        }
+    }
+
+    fn get_status_internal(&self) -> Result<Vec<Result<JenkinsBuildStatus, Error>>, Error> {
```

Comment:
This `get_status_internal` was moved here from `main.rs`, and used to be called `run_one_jenkins`. It underwent some very minor changes in the process.